### ISSUE TITLE
[Bug Report] pytest is not a dev dependency

### ISSUE THREAD

**Issue by JasonGross** [neelnanda-io/TransformerLens#503]
If you are submitting a bug report, please fill in the following details and use the tag [bug].

**Describe the bug**
pytest is listed as a dev dependency at
https://github.com/neelnanda-io/TransformerLens/blob/829084a53836c5b8b388aa37a5ffce73b6371712/pyproject.toml#L37-L47
but it is used in utils at
https://github.com/neelnanda-io/TransformerLens/blob/829084a53836c5b8b388aa37a5ffce73b6371712/transformer_lens/utils.py#L16
and `import transformer_lens` fails if pytest is not installed.

**Code example**
```python
import transformer_lens
```
gives
```
Traceback (most recent call last):
  File "/Users/rajashreeagrawal/Directory/2024Directory/sparse_autoencoder_playground/sae.py", line 16, in <module>
    import transformer_lens
  File "/Users/rajashreeagrawal/Directory/2024Directory/sparse_autoencoder_playground/.venv/lib/python3.10/site-packages/transformer_lens/__init__.py", line 2, in <module>
    from . import utils
  File "/Users/rajashreeagrawal/Directory/2024Directory/sparse_autoencoder_playground/.venv/lib/python3.10/site-packages/transformer_lens/utils.py", line 16, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'
```

**System Info**
Describe the characteristic of your environment:
 * Describe how `transformer_lens` was installed (pip, docker, source, ...) `poetry add transformer_lens`
 * What OS are you using? (Linux, MacOS, Windows)  MacOS
 * Python version (We suppourt 3.7 -3.10 currently) 3.10

**Additional context**

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

**Comment by stprior** [comment]
I'm seeing this too. Looks like it will be fixed by PR #463

**Comment by stprior** [comment]
Replicated this with a clean python environment using `pip install transformer_lens`

Looks like it was introduced in #463 

Rather than adding a pytest dependency should the change to utils.py just be reverted?
